src/java.base/share/classes/jdk/internal/foreign/MemorySessionImpl.java

```diff
@@ -277,7 +277,11 @@ static ResourceCleanup ofRunnable(Runnable cleanupAction) {
                 return new ResourceCleanup() {
                     @Override
                     public void cleanup() {
-                        cleanupAction.run();
+                        try {
+                            cleanupAction.run();
+                        } catch (Throwable ex) {
+                            // swallow
```

Comment:
> @tprinzing This may be an area that you want to look at sometime to see if a JFR event might make sense.

Might make sense - I suggest that should be done also for Cleaner - as that's using same mechanism (and no JFR events)